jobs/extractor.py

```diff
@@ -0,0 +1,61 @@
+import requests 
+import pandas as pd
+from loader import S3Loader
+
+class SWAPIClient:
+    def __init__(self):
+        # Constrói a URL completa para o endpoint especificado
+        self.base_url = f"https://swapi.dev/api/"
+    # Função para extrair dados de um endpoint específico com paginação
+    def extract_data(self, endpoint):
+        # TO-DO: colocar os logs nas etapas de requisição
+        url = self.base_url + endpoint
+        # Cria uma uma sessão em cada interação, permitindo que a conexão seja mantida aberta
+        ## Envia uma solicitação GET para a URL usando a biblioteca 'requests'
+        session = requests.Session()
+        response = session.get(url)
+        # Verifica se a reposta tem um status code 200 (indicando uma resposta bem sucedida)
+        if response.status_code == 200:
+            # Extrai os dados da resposta em formato JSON
+            data = response.json()
+            # Inicializa uma lista chamada 'results' com os resultados extraídos 
+            results = data['results']
+            # Verifica se há um link para a próxima página nos dados extraídos
+            while data['next']:
+                # Havendo próxima página, envia uma nova solicitação GET 
+                response = session.get(data['next'])
+                data = response.json()
+                # Adiciona os dados da próxima página à lista 'results'
+                results.extend(data['results'])
+            # Retorna a lista completa de resultados extraídos
+            df = pd.DataFrame(results)
+            df.replace(to_replace=[r"\t|\n|\r"], value=[""], regex=True, inplace=True)
```

Comment:
Está errado! Minha sugestão:
```suggestion
            df = df.replace(to_replace=[r"\t|\n|\r"], value=[""], regex=True)
```